packages/cardano-services/src/StakePool/HttpStakePoolMetadata/HttpStakePoolMetadataService.ts

```diff
@@ -0,0 +1,170 @@
+        // Validate CIP-6 ext metadata fields
+        if (metadata.extDataUrl && (!metadata.extSigUrl || !metadata.extVkey)) {
+          return {
+            errors: [
+              new StakePoolMetadataServiceError(
+                StakePoolMetadataServiceFailure.InvalidMetadata,
+                null,
+                'Missing ext signature or public key'
+              )
+            ],
+            metadata
+          };
+        }
+
+        // Fetch extended metadata (supports both cip-6 and ada pools formats already)
+        extMetadata = await this.getStakePoolExtendedMetadata(metadata);
+
+        // In case of CIP-6 standard -> perform signature verification
+        if (metadata.extDataUrl && metadata.extSigUrl && metadata.extVkey) {
+          // Based on the CIP-6, we have `extSigUrl` (A URL with the extended metadata signature), so we need to make another HTTP request to get the actual signature
+          try {
+            const signature = (await axiosClient.get<Crypto.Ed25519SignatureHex>(metadata.extSigUrl)).data;
+            const message = HexBlob.fromBytes(Buffer.from(JSON.stringify(extMetadata)));
+            const publicKey = Crypto.Ed25519PublicKeyHex(metadata.extVkey);
+            const bip32Ed25519 = new Crypto.CmlBip32Ed25519(CML);
+
+            // Verify the signature
+            const isSignatureValid = await bip32Ed25519.verify(signature, message, publicKey);
+
+            // If not valid -> omit extended metadata from response and add specific error
+            if (!isSignatureValid) {
+              extMetadata = undefined;
+              errors.push(
+                new StakePoolMetadataServiceError(
+                  StakePoolMetadataServiceFailure.InvalidExtendedMetadataSignature,
+                  null,
+                  'Invalid extended metadata signature'
+                )
+              );
+            }
+
+            // If signature url failed -> omit extended metadata from response and add specific error
+          } catch {
+            extMetadata = undefined;
+            errors.push(
+              new StakePoolMetadataServiceError(
+                StakePoolMetadataServiceFailure.FailedToFetchExtendedSignature,
+                null,
+                `${SERVICE_NAME} failed to fetch extended signature from ${metadata.extSigUrl} due to connection error`
+              )
+            );
+          }
+        }
+      }
+    } catch (error) {
+      if (axios.isAxiosError(error)) {
+        errors.push(
+          new StakePoolMetadataServiceError(
+            StakePoolMetadataServiceFailure.FailedToFetchMetadata,
+            error.toJSON(),
+            `${SERVICE_NAME} failed to fetch metadata JSON from ${url} due to ${error.message}`
+          )
+        );
+      } else if (error instanceof StakePoolMetadataServiceError) {
+        errors.push(new StakePoolMetadataServiceError(error.reason, error.innerError, error.detail));
```

Comment:
Is there a reason to clone here?
```suggestion
        errors.push(error);
```

Reply:
I think we just can push the thrown error